File: CPP/TBoard.cpp
// TBoard.cpp: implementation of the TBoard class.
//
//////////////////////////////////////////////////////////////////////

#include "stdafx.h"
#include "TBoard.h"

//////////////////////////////////////////////////////////////////////
// Construction/Destruction
//////////////////////////////////////////////////////////////////////

TBoard::TBoard(const int Ht, const int Wt)
{
	int i, j;
//	srand(3);
    bHt = Ht;
    bWt = Wt;
	board = new int* [Ht];

	for(i = 0; i < Ht; i++)
	{
		board[i] = new int [Wt];
		for(j = 0; j < Wt; j++)
		{
			board[i][j] = 5.0 * rand() / RAND_MAX + 1;
			if( i == Ht - 1 && j%2) board[i][j] = 0;
		}
	}
}

TBoard::~TBoard()
{
	for(int i = 0; i < bHt; i++)
		delete[] board[i];
	delete[] board;
}

int TBoard::GetCell(int i, int j)
{
	return board[i][j];
}

int TBoard::SetCell(int i, int j, int playerCD)
{
	board[i][j] += playerCD;
	return board[i][j];
}

int TBoard::ResetCell(int i, int j, int cell)
{
	board[i][j] = cell;
	return board[i][j];
}

File: src/game/TBoard.ts
export class TBoard {
    private board: number[][];
    private bHt: number;
    private bWt: number;

    constructor(Ht: number, Wt: number) {
        this.bHt = Ht;
        this.bWt = Wt;
        this.board = [];

        // Initialize board matching the original C++ logic
        for (let i = 0; i < Ht; i++) {
            this.board[i] = [];
            for (let j = 0; j < Wt; j++) {
                // Match the original logic: 5.0 * rand() / RAND_MAX + 1
                this.board[i][j] = Math.floor(5.0 * Math.random()) + 1;
                if (i === Ht - 1 && j % 2) {
                    this.board[i][j] = 0;
                }
            }
        }
    }

    getCell(i: number, j: number): number {
        return this.board[i][j];
    }

    setCell(i: number, j: number, playerCD: number): number {
        this.board[i][j] += playerCD;
        return this.board[i][j];
    }

    resetCell(i: number, j: number, cell: number): number {
        this.board[i][j] = cell;
        return this.board[i][j];
    }

    getH(): number {
        return this.bHt;
    }

    getW(): number {
        return this.bWt;
    }

    // Method to get the entire board for debugging/display
    getBoard(): number[][] {
        return this.board.map(row => [...row]);
    }
} 